naxsi_src/naxsi_skeleton.c

```diff
@@ -1159,3 +1272,473 @@ static ngx_int_t ngx_http_dummy_access_handler(ngx_http_request_t *r)
+    if (!tmp_uri->data) {
+        return (NGX_ERROR);
+    }
+    ngx_escape_uri(tmp_uri->data, r->uri.data, r->uri.len, NGX_ESCAPE_ARGS);
+
+    v->data = tmp_uri->data;
+    v->len = tmp_uri->len;
+    v->valid = 1;
+    v->no_cacheable = 0;
+    v->not_found = 0;
+    return NGX_OK;
+}
+
+
+static ngx_int_t
+ngx_http_naxsi_learning_variable(ngx_http_request_t *r, ngx_http_variable_value_t *v, uintptr_t data)
+{
+    ngx_http_request_ctx_t *ctx = recover_request_ctx(r);
+    if (!ctx) {
+        v->not_found = 1;
+        return NGX_OK;
+    }
+
+    v->data = ngx_palloc(r->pool, 1);
+    if (v->data == NULL) {
+        return NGX_ERROR;
+    }
+    v->data[0] = ctx->learning ? '1' : '0';
+    v->len = 1;
+    v->valid = 1;
+    v->no_cacheable = 0;
+    v->not_found = 0;
+
+    return NGX_OK;
+}
+
+
+static ngx_int_t
+ngx_http_naxsi_block_variable(ngx_http_request_t *r, ngx_http_variable_value_t *v, uintptr_t data)
+{
+    ngx_http_request_ctx_t *ctx = recover_request_ctx(r);
+    if (!ctx) {
+        v->not_found = 1;
+        return NGX_OK;
+    }
+
+    v->data = ngx_palloc(r->pool, 1);
+    if (v->data == NULL) {
+        return NGX_ERROR;
+    }
+    v->data[0] = ctx->block ? '1' : '0';
+    v->len = 1;
+    v->valid = 1;
+    v->no_cacheable = 0;
+    v->not_found = 0;
+
+    return NGX_OK;
+}
+
+
+static ngx_int_t
+ngx_http_naxsi_total_processed_variable(ngx_http_request_t *r, ngx_http_variable_value_t *v, uintptr_t data)
+{
+    ngx_http_dummy_loc_conf_t *cf = ngx_http_get_module_loc_conf(r, ngx_http_naxsi_module);
+
+    v->data = ngx_palloc(r->pool, NGX_INT32_LEN);
+    if (v->data == NULL) {
+        return NGX_ERROR;
+    }
+    v->len = ngx_sprintf(v->data, "%z", cf->request_processed) - v->data;
+    v->valid = 1;
+    v->no_cacheable = 0;
+    v->not_found = 0;
+
+    return NGX_OK;
+}
+
+
+static ngx_int_t
+ngx_http_naxsi_total_blocked_variable(ngx_http_request_t *r, ngx_http_variable_value_t *v, uintptr_t data)
+{
+    ngx_http_dummy_loc_conf_t *cf = ngx_http_get_module_loc_conf(r, ngx_http_naxsi_module);
+
+    v->data = ngx_palloc(r->pool, NGX_INT32_LEN);
+    if (v->data == NULL) {
+        return NGX_ERROR;
+    }
+    v->len = ngx_sprintf(v->data, "%z", cf->request_blocked) - v->data;
+    v->valid = 1;
+    v->no_cacheable = 0;
+    v->not_found = 0;
+
+    return NGX_OK;
+}
+
+
+static ngx_int_t
+ngx_http_naxsi_score_variable(ngx_http_request_t *r, ngx_http_variable_value_t *v, uintptr_t data)
+{
+    ngx_http_request_ctx_t *ctx = recover_request_ctx(r);
+    if (!ctx) {
+        v->not_found = 1;
+        return NGX_OK;
+    }
+
+    const char *fmt = "%d:%.*s:%d,"; /* id:cscore:score */
+    ngx_http_matched_rule_t *mr;
+    ngx_http_special_score_t *sc;
+    ngx_uint_t others = 0;
+    ngx_uint_t i;
+    size_t sz = 0;
+    char* p;
+
+    if (ctx->matched) {
+        mr = ctx->matched->elts;
+        for (i = 0; i < ctx->matched->nelts; i++) {
+            if (mr[i].rule->rule_id < 1000) {
```

Comment:
Yes, a simple comments indicating ids < 1000 are internal rules is good for clarity :)